### PR TITLE
enh(mc): Refreshed Azure Functions page

### PR DESCRIPTION
**Pull request by omercier**
## Description

Refreshed Azure Functions page

## Target version (i.e. version that this PR changes)

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] Cloud
- [x] Monitoring Connectors

### CODE REVIEW

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 1]
```diff
@@ -5,412 +5,866 @@ title: Azure Functions
...
+| --warning-*      | Warning threshold where '*' can be: 'connections', 'assemblies', 'handle', 'thread', 'app-domains', 'app-domain-unloaded'.      |
+| --critical-*     | Critical threshold where '*' can be:. 'connections', 'assemblies', 'handle', 'thread', 'app-domains', 'app-domain-unloaded'.    |
 
-#### Les identifiants ont changé et mon Plugin ne fonctionne plus
+</TabItem>
+<TabItem value="Cpu-Time" label="Cpu-Time">
+
+| Option              | Description                                                 |
+|:--------------------|:------------------------------------------------------------|
+| --resource          | Set resource name or id (Required).                         |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:751]:
```suggestion
| --resource          | Set resource name or ID (required).                         |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 2]
```diff
@@ -5,412 +5,866 @@ title: Azure Functions
...
+| --resource-group    | Set resource group (Required if resource's name is used).   |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:752]:
```suggestion
| --resource-group    | Set resource group (required if resource's name is used).   |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 3]
```diff
@@ -5,412 +5,866 @@ title: Azure Functions
...
+
+Les options disponibles pour chaque modèle de services sont listées ci-dessous :
+
+<Tabs groupId="sync">
+<TabItem value="App-Usage" label="App-Usage">
+
+| Option           | Description                                                                                                                     |
+|:-----------------|:--------------------------------------------------------------------------------------------------------------------------------|
+| --resource       | Set resource name or id (Required).                                                                                             |
+| --resource-group | Set resource group (Required if resource's name is used).                                                                       |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:742]:
```suggestion
| --resource-group | Set resource group (required if resource's name is used).                                                                       |
```

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 4]
```diff
@@ -5,412 +5,866 @@ title: Azure Functions
...
+| Data          | Cloud-Azure-Compute-Functions-Data-Api-custom          | Contrôle l'utilisation de la bande passante de l'application Azure Functions       |
+| Executions    | Cloud-Azure-Compute-Functions-Executions-Api-custom    | Contrôle le nombre d'exécutions de fonctions                                       |
+| File-System   | Cloud-Azure-Compute-Functions-FIle-System-Api-custom   | Contrôle l'utilisation du système de fichiers de l'application Azure Functions     |
+| Gc-Usage      | Cloud-Azure-Compute-Functions-Gc-Usage-Api-custom      | Contrôle l'utilisation du récupérateur de mémoire de l'application Azure Functions |
+| Health        | Cloud-Azure-Compute-Functions-Health-Api-custom        | Contrôle la disponibilité des ressources Azure Functions                           |
+| Http-Requests | Cloud-Azure-Compute-Functions-Http-Requests-Api-custom | Contrôle les requêtes HTTP de l'application Azure Functions                        |
+| IO-Operations | Cloud-Azure-Compute-Functions-IO-Operations-Api-custom | Contrôle les opération E                                                           |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:39]:
```suggestion
| IO-Operations | Cloud-Azure-Compute-Functions-IO-Operations-Api-custom | Contrôle les opération E/S de l'application Azure Functions.                                                           |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 5]
```diff
@@ -5,412 +5,866 @@ title: Azure Functions
...
 
-| Metric Name                          | Description                                                                       | Unit  |
-|:-------------------------------------|:----------------------------------------------------------------------------------|:------|
-| appservice.connections.count         | The number of bound sockets existing in the sandbox                               | Count |
-| appservice.assemblies.current.count  | The current number of assemblies loaded across all AppDomains in this application | Count |
-| appservice.handle.count              | The total number of handles currently open by the app process                     | Count |
-| appservice.thread.count              | The number of threads currently active in the app process                         | Count |
-| appservice.appdomains.count          | The current number of AppDomains loaded in this application                       | Count |
-| appservice.appdomains.unloaded.count | The total number of AppDomains unloaded since the start of the application        | Count |
+| Métrique                             | Unité |
```
[the remark this answers is not in the document]
Comment (omercier) [reply i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:69]: Parce que ce n'est plus dans le template ?

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (omercier) [reply i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md]: L'originale est `Check Azure App Service I/O operations by the app.`

File: pp/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 6]
```diff
@@ -5,402 +5,865 @@ title: Azure Functions
...
+| App-Usage     | Cloud-Azure-Compute-Functions-App-Usage-Api-custom     | Check Azure Functions app usage                    |
+| Cpu-Time      | Cloud-Azure-Compute-Functions-Cpu-Time-Api-custom      | Check Azure Functions Cpu time consumed by the app |
+| Data          | Cloud-Azure-Compute-Functions-Data-Api-custom          | Check Azure Functions app bandwith usage           |
+| Executions    | Cloud-Azure-Compute-Functions-Executions-Api-custom    | Check functions executions                         |
+| File-System   | Cloud-Azure-Compute-Functions-FIle-System-Api-custom   | Check Azure Functions file system                  |
+| Gc-Usage      | Cloud-Azure-Compute-Functions-Gc-Usage-Api-custom      | Check Azure Functions garbage collector usage      |
+| Health        | Cloud-Azure-Compute-Functions-Health-Api-custom        | Check Azure Functions                              |
+| Http-Requests | Cloud-Azure-Compute-Functions-Http-Requests-Api-custom | Check Azure Functions HTTP requests                |
+| IO-Operations | Cloud-Azure-Compute-Functions-IO-Operations-Api-custom | Check Azure Functions I                            |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:37]:
```suggestion
| IO-Operations | Cloud-Azure-Compute-Functions-IO-Operations-Api-custom | Check Azure Functions I/O operations.                            |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 7]
```diff
@@ -5,412 +5,866 @@ title: Azure Functions
...
+	--client-id='xxxxxxxxx' \
+	--client-secret='xxxxxxxxx' \
+	--proxyurl=''  \
+	--filter-metric='' \
+	--filter-dimension='' \
+	--timeframe='900' \
+	--interval='PT5M' \
+	--aggregation='Average' \
+	--warning-status='' \
+	--critical-status=''
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:563]:
```suggestion
    --plugin=cloud::azure::compute::functions::plugin \
    --mode=cpu-time \
    --custommode=api \
    --subscription='xxxxxxxxx' \
    --tenant='xxxxxxxxx' \
    --client-id='xxxxxxxxx' \
    --client-secret='xxxxxxxxx' \
    --resource='APP01' \
    --resource-group='xxxxxxxxx' \
    --timeframe='900' \
    --interval='PT5M' \
    --aggregation='Total' \
    --warning-cpu-time='1' \
    --critical-cpu-time='2'
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 8]
```diff
@@ -5,412 +5,866 @@ title: Azure Functions
...
+	--warning-status='' \
+	--critical-status='' 
 ```
 
-La commande devrait retourner un message de sortie similaire à:
+La commande devrait retourner un message de sortie similaire à :
 
 ```bash
-OK: Instance 'APP01' Statistic 'total' Metrics CPU Time: 0.08s | 'APP01~total#appservice.cpu.consumed.seconds'=0.08s;;;0;
+OK: Health check status: 96 | 'appservice.status.count'=96;;;0;
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:569]:
```suggestion
OK: Instance 'APP01' Statistic 'total' Metrics CPU Time: 0.08s | 'APP01~total#appservice.cpu.consumed.seconds'=0.08s;;;0;
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 9]
```diff
@@ -5,412 +5,866 @@ title: Azure Functions
...
 ```
 
-La commande ci-dessus vérifie le temps processeur consommé sur l'instance *Functions* nommée *APP01*
-(```--plugin=cloud::azure::compute::functions::plugin --mode=cpu-time --resource='APP01'```).
+### Diagnostic des erreurs communes
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:572]:
```suggestion
La commande ci-dessus vérifie le temps processeur consommé sur l'instance *Functions* nommée *APP01*
(```--plugin=cloud::azure::compute::functions::plugin --mode=cpu-time --resource='APP01'```).

Le mode de connexion utilisé est 'api' (```--custommode=api```), les paramètres d'authentification nécessaires à l'utilisation de ce mode
sont donc renseignés en fonction (```--subscription='xxxxxxxxx' --tenant='xxxxxxx' --client-id='xxxxxxxx' --client-secret='xxxxxxxxxx'```).

Les statuts caculés se baseront sur les valeurs totales d'un échantillon dans un intervalle de 15 minutes / 900 secondes  (```--timeframe='900'```)
avec un état retourné par tranche de 5 minutes (```--interval='PT5M'```).

Dans cet exemple, une alarme de type WARNING sera déclenchée si le temps de processeur consommé pendant l'intervalle donné
est supérieur à 1 (```--warning-cpu-time='1'```); l'alarme sera de type CRITICAL au-delà de 2
(```--critical-cpu-time='2'```).

### Diagnostic des erreurs communes
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 10]
```diff
@@ -5,402 +5,863 @@ title: Azure Functions
...
+	--client-id='' \
+	--client-secret='' \
+	--proxyurl=''  \
+	--filter-metric='' \
+	--filter-dimension='' \
+	--timeframe='900' \
+	--interval='PT5M' \
+	--aggregation='Average' \
+	--warning-status='' \
+	--critical-status=''
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:560]:
```suggestion
    --plugin=cloud::azure::compute::functions::plugin \
    --mode=cpu-time \
    --custommode=api \
    --subscription='xxxxxxxxx' \
    --tenant='xxxxxxxxx' \
    --client-id='xxxxxxxxx' \
    --client-secret='xxxxxxxxx' \
    --resource='APP01' \
    --resource-group='xxxxxxxxx' \
    --timeframe='900' \
    --interval='PT5M' \
    --aggregation='Total' \
    --warning-cpu-time='1' \
    --critical-cpu-time='2'
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 11]
```diff
@@ -5,402 +5,863 @@ title: Azure Functions
...
+	--warning-status='' \
+	--critical-status='' 
 ```
 
-Expected command output is shown below:
+The expected command output is shown below:
 
 ```bash
-OK: Instance 'APP01' Statistic 'total' Metrics CPU Time: 0.08s | 'APP01~total#appservice.cpu.consumed.seconds'=0.08s;;;0;
+OK: Health check status: 96 | 'appservice.status.count'=96;;;0;
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:566]:
```suggestion
OK: Instance 'APP01' Statistic 'total' Metrics CPU Time: 0.08s | 'APP01~total#appservice.cpu.consumed.seconds'=0.08s;;;0;
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md [hunk 12]
```diff
@@ -5,402 +5,863 @@ title: Azure Functions
...
 ```
 
-The command above checks the *Cpu-Time* of an Azure *Functions* instance using the 'api' custom-mode
-(```--plugin=cloud::azure::compute::functions::plugin --mode=cpu-time --custommode=api```).
-This Functions instance is identified by its id (```--resource='APP01'```) and the authentication parameters
-to be used with the custom mode are specified in the options (```--subscription='xxxxxxxxx' --tenant='xxxxxxx'
---client-id='xxxxxxxx' --client-secret='xxxxxxxxxx'```).
+### Troubleshooting
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/cloud-azure-compute-functions.md:569]:
```suggestion

The command above checks the *Cpu-Time* of an Azure *Functions* instance using the 'api' custom-mode
(```--plugin=cloud::azure::compute::functions::plugin --mode=cpu-time --custommode=api```).
This Functions instance is identified by its id (```--resource='APP01'```) and the authentication parameters
to be used with the custom mode are specified in the options (```--subscription='xxxxxxxxx' --tenant='xxxxxxx'
--client-id='xxxxxxxx' --client-secret='xxxxxxxxxx'```).

The calculated metrics are the total (```--aggregation='Total'```) of values on a 900 secondes / 15 min period (```--timeframe='900'```)
with one sample per 5 minutes (```--interval='PT5M'```).

This command would trigger a WARNING alarm if the *Cpu-time* is reported as over 1
(```--warning-cpu-time='1'```) and a CRITICAL alarm over 2 (```--critical-cpu-time='2'```).

### Troubleshooting
```

**Review by cg-tw — APPROVED** [review]

**Pull request merged**